wgsl/index.bs

```diff
@@ -7347,13 +7345,344 @@ See [[#expressions]].
 Statements in a [SHORTNAME] program are executed in control flow order.
 See [[#statements]] and [[#function-calls]].
 
-## Uniformity TODO ## {#uniformity}
+## Uniformity ## {#uniformity}
+
+### Terminology and concepts ### {#uniformity-concepts}
+
+The following definitions are merely informative, trying to give an intuition for what the analysis in the next subsection is computing.
+The analysis is what actually defines these concepts, and when a program is valid or breaks the uniformity rules.
+
+Control flow is said to be uniform at a point in the program if all invocations execute in lockstep at that program point.
+An expression is said to be uniform if it is in uniform control flow, and all invocations compute the same value for it.
+A local variable is said to be uniform, if all invocations have the same value in that variable at every program point where it is live.
+
+### Uniformity analysis overview ### {#uniformity-overview}
+
+In this section we specify an analysis that verifies that functions which are only safe to call in uniform control flow (e.g. barriers and derivatives) are only called in such a context.
```

Comment:
Perhaps we could refactor this to say, first, that there are special functions that require uniformity of control flow?
Then we'd say that this section describes the analysis done to figure out if calls to them are valid.

Reply:
I've rephrased this sentence along those lines.